Library/Homebrew/cmd/--selfdestruct.rb

```diff
@@ -0,0 +1,93 @@
+# Original Inspiration: Stephen Benner https://github.com/SteveBenner: https://gist.github.com/SteveBenner/11254428
+# Partially Rewritten by Dominyk Tiller https://github.com/DomT4 for Homebrew's internal use.
+# THIS WILL COMPLETELY REMOVE HOMEBREW FROM YOUR SYSTEM
+
+require "fileutils"
+require "open3"
+require "rubygems"
+
+begin
+  require "trollop"
+rescue LoadError
+  onoe "You must run `sudo /System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/gem install trollop`"
+  raise
+end
```

Comment:
Yes, for now. This is 90% because I found trollop the other day and played with it, and really loved the interface. It's clean, easy and reliable. There's probably an internal Ruby requirement we could use if necessary... I'll just need to hunt it down. We could also lessen the permanence of the gem by installing the gem into ` /tmp ` and then telling Brew to look there for it for the duration of the script.